Backend/Remora.Discord.Core/IOptional.cs

```diff
@@ -33,6 +34,17 @@ public interface IOptional
         /// <summary>
         /// Gets a value indicating whether the optional contains a value.
         /// </summary>
+        [Obsolete("Prefer IOptional.IsSpecified")]
         bool HasValue { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the optional value is logically present.
+        /// </summary>
+        bool IsSpecified { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the value is <c>null</c>. If <see cref="IsSpecified"/> is false, <see cref="IsNull"/> is undefined.
+        /// </summary>
+        bool IsNull { get; }
```

Comment:
An `Optional<T>` is not concerned with whether its contained value is null (or, in fact, if it is nullable at all). It is up to the declaring and using individual to appropriately check things based on the generic type. I don't see a compelling reason to add this property.

Reply:
This might be a useful extension property to write for `Optional<T?> where T : struct` and `Optional<T?> where T : class` types, but even then, it probably doesn't belong in Remora itself.